### PR TITLE
Fix version path traversal

### PR DESCRIPTION
**Pull request by ravi-kumar-pilla**
## Description

Added checks to validate safe versions

## Development notes

- Added a check for filtering unsafe versions
- Raises respective error messages incase of an unsafe version
- Updated tests 

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: kedro/utils.py [hunk 1]
```diff
@@ -311,3 +311,19 @@ def find_config_file(
...
+
+def _is_unsafe_version(version: str) -> bool:
+    """Return True if the version string contains path traversal or absolute path components.
+
+    Checks against both POSIX and Windows path semantics to ensure platform-independent
+    protection against directory traversal attacks.
+    """
+    posix = PurePosixPath(version)
+    windows = PureWindowsPath(version)
+    return (
```
Comment (ElenaKhaustova) [inline kedro/utils.py]: The issue description says that `path separators used as directory components` should be rejected. Currently, versions like `foo/bar` pass validation and create unexpected subdirectory structures. Also, `"." ` resolves to the parent path, which can cause confusing behaviour.

Shall we cover these cases as well?

- Empty strings `""`
- Forward slashes `"foo/bar"`
- Backslashes `"foo\bar"`
- Single dot `"."` which resolves `filepath/./name` to `filepath/name`

File: kedro/framework/cli/utils.py [hunk 2]
```diff
@@ -503,7 +503,13 @@ def _split_load_versions(ctx: click.Context, param: Any, value: str) -> dict[str
                 f"'dataset_name:YYYY-MM-DDThh.mm.ss.sssZ',"
                 f"found {load_version} instead"
             )
-        load_versions_dict[load_version_list[0]] = load_version_list[1]
+        version = load_version_list[1]
+        if _is_unsafe_version(version):
+            raise KedroCliError(
+                f"Version string '{version}' is not allowed. "
```
Comment (ElenaKhaustova) [inline kedro/framework/cli/utils.py:509]: Shall we also update error message text to mention path separators, since that should also be rejected (see comment below)?

**Review by ElenaKhaustova — COMMENTED** [review]
Thank you, @ravi-kumar-pilla!

It looks mostly good, but I think we might be missing some cases mentioned in the original issue description (see the comment).

Could you please double-check that, and if so, extend the missing edge-case tests as well?

File: tests/framework/cli/test_cli.py [hunk 3]
```diff
@@ -1020,6 +1031,30 @@ def test_fail_split_load_versions(self, fake_project_cli, fake_metadata):
...
+    )
+    def test_fail_split_load_versions_path_traversal(
+        self, fake_project_cli, fake_metadata, unsafe_version
+    ):
+        result = CliRunner().invoke(
+            fake_project_cli,
+            ["run", "--load-versions", unsafe_version],
+            obj=fake_metadata,
+        )
+        assert result.exit_code
```
Comment (ElenaKhaustova) [inline tests/framework/cli/test_cli.py:1055]:
```suggestion
        assert result.exit_code == 2
```

Comment (ravi-kumar-pilla) [reply tests/framework/cli/test_cli.py:1055]: Hi @ElenaKhaustova , thanks for the review. The exit_code is 1 and not 2 similar to - https://github.com/kedro-org/kedro/blob/main/tests/framework/cli/test_cli.py#L1014

Comment (ElenaKhaustova) [reply tests/framework/cli/test_cli.py:1055]: Good point, you're right.
